JavaScript/JavaScriptSDK.Tests/CheckinTests/Context/Sample.tests.ts

```diff
@@ -0,0 +1,133 @@
+﻿/// <reference path="../../../JavaScriptSDK/appInsights.ts" />
+/// <reference path="../../../JavaScriptSDK/context/sample.ts" />
+/// <reference path="../../testframework/common.ts" />
+/// <reference path="../Util.tests.ts"/>
+
+class SampleContextTests extends TestClass {
+
+    private originalDocument;
+    private results: any[];
+
+    /** Method called before the start of each test method */
+    public testInitialize() {
+        this.results = [];
+    }
+
+    /** Method called after each test method has completed */
+    public testCleanup() {
+        this.results = [];
+    }
+
+    public registerTests() {
+        var contextKeys = new AI.ContextTagKeys();
+
+        this.testCase({
+            name: "Sampling: isSampledIn returns true for 100 sampling rate",
+            test: () => {
+                // setup
+                var sample = new Microsoft.ApplicationInsights.Context.Sample(100);
+
+                var envelope1 = this.GetEnvelope();
+                envelope1.tags[contextKeys.userId] = null;
+                envelope1.tags[contextKeys.operationId] = null;
+                                
+                // act
+                var isSampledIn1 = sample.IsSampledIn(envelope1);
+
+                // assert
+                Assert.ok(isSampledIn1);
+            }
+        });
+
+        this.testCase({
```

Comment:
One more test worth adding would be to check what happens if a sample with a illegal sample value is created.